staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -20,30 +20,47 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"encoding/base64"
+	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
+	"net/http"
+	"net/url"
 	"os"
 	"time"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/wait"
 	apiserverconfig "k8s.io/apiserver/pkg/apis/config"
 	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
 	"k8s.io/apiserver/pkg/storage/value"
 	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/envelope"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/secretbox"
+	"k8s.io/klog"
 )
 
 const (
-	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
-	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
-	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1:"
-	kmsTransformerPrefixV1       = "k8s:enc:kms:v1:"
-	kmsPluginConnectionTimeout   = 3 * time.Second
+	aesCBCTransformerPrefixV1      = "k8s:enc:aescbc:v1:"
+	aesGCMTransformerPrefixV1      = "k8s:enc:aesgcm:v1:"
+	secretboxTransformerPrefixV1   = "k8s:enc:secretbox:v1:"
+	kmsTransformerPrefixV1         = "k8s:enc:kms:v1:"
+	kmsPluginConnectionTimeout     = 3 * time.Second
+	kmsPluginHealthzTimeout        = 90 * time.Second
+	kmsPluginHealthzRequestTimeout = 3 * time.Second
+	kmsPluginHealthzRetryInterval  = 5 * time.Second
+)
+
+var (
+	errKMSPluginUnhealthy                = errors.New("kms-plugin's is unhealthy")
+	errInvalidHealthzTimeoutValue        = errors.New("value of HealthzEndpoint.Timeout should be a positive")
```

Comment:
nit, use serialized field names, since that is what the user will have to use to set these values